coap-gateway/service/client.go

```diff
@@ -167,7 +168,7 @@ func (client *Client) onObserveResource(ctx context.Context, deviceID, href stri
 		return fmt.Errorf("cannot handle resource observation: %w", err)
 	}
 	notification.Hijack()
-	err := client.server.taskQueue.Submit(func() {
+	err := client.server.taskQueue.SubmitForOneWorker(resource.GetInstanceID(deviceID+href), func() {
```

Comment:
Small optimization: we could calculate the `instanceID` value when observation is created and sent it to the callback as a parameter instead of always calculating it here in the callback.

Reply:
`resource.GetInstanceID` is very cheap function. So I don't want to optimize.